### PR TITLE
feat(routes/all): provide meta, refactor response payload

### PR DESCRIPTION
**Pull request by tthew**
This commit changes the way the routes/all handler behaves.

The handler now makes two parallel calls to the couchdb. One to ascertain
the total length of the data set, and one to run the actual query that
was provided to the api call and return the relevant set and returns
both to the client as a json object, as usual. However;

---

BREAKING CHANGE:

In order to return this additional meta data we need to modify the
strucure of the json object returned by the handler. Now instead of
returning an array containing the result set we return an object, with 
a `total` property containing the total length of the data set and a 
`data` property that contains the result set.

```diff
diff --git a/2.json b/1.json
index 0269e50..7dd8c5f 100644
--- a/2.json
+++ b/1.json
@@ -1,14 +1,17 @@
-[
-   {
+{
+  total: 100,
+  data: [
+    {
       "name":"Name",
       "location":{
-         "place":"Place",
+        "place":"Place",
       },
       "id":"1"
-   }
-]
+    }
+  ]
+}
```

### CODE REVIEW

File: tests/spec/routes.js [hunk 1]
```diff
@@ -68,7 +71,7 @@ test('Entities API', function (t) {
       }
     }, function (response) {
       t.is(response.statusCode, 200, 'should respond 200')
-      t.is(response.result.length, 1, 'should return just one item')
+      t.is(response.result.data.length, 1, 'should return just one item')
```
Comment (karl-ehealth) [inline tests/spec/routes.js]: can we have a test for total here as well? would be nice to demostrate the functionality :)

**Comment by tthew** [comment]
@gr2m thanks for the tip, fixed up.  Any thoughts on this?

**Comment by gr2m** [comment]
yeah I think it makes sense. Eventually this would be a great candidate for JSON API, but it would cause quite some overhead to migrate everything.

**Comment by karl-ehealth** [comment]
LGTM!

**Pull request merged**